### PR TITLE
feature(octopoes): fields to finding type model

### PR DESCRIPTION
**Pull request by Lisser**
### Changes
Adds fields to FindingType model

### Issue link
Implements part 1 of:  
https://github.com/minvws/nl-kat-coordination/issues/653

### Proof
N/A

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: octopoes/octopoes/models/ooi/findings.py [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
-class FindingType(OOI):
+class RiskLevelSeverity(Enum):
+    CRITICAL = "critical"
+    HIGH = "high"
+    MEDIUM = "medium"
+    LOW = "low"
+    RECOMMENDATION = "recommendation"
+
+
+class FindingType(OOI, abc.ABC):
```
Comment (dekkers) [inline octopoes/octopoes/models/ooi/findings.py:19]: Why inherit from `abc.ABC`? This on itself doesn't really add anything. `abc.ABC` has very specific functionality that we don't seem to be using, so I don't really see the point of inheriting from `abc.ABC`.

Comment (Lisser) [reply octopoes/octopoes/models/ooi/findings.py:19]: @dekkers 

`FindingType` is an abstract class, so it should never be instantiated directly. So, IMO better to enforce it?

Comment (dekkers) [reply octopoes/octopoes/models/ooi/findings.py:19]: But that is not what `abc.ABC` does. See for example:

```
In [1]: import abc

In [2]: class Test(abc.ABC):
   ...:     pass
   ...: 

In [3]: a = Test()

In [4]: print(a)
<__main__.Test object at 0x7f5423fba4d0>
```

Comment (ammar92) [reply octopoes/octopoes/models/ooi/findings.py:19]: Good points. Technically it doesn't add anything, semantically it has meaning. I don't really have a strong opinion on this, but I'd suggest leaving it out as it's not supposed to be used like this.

File: octopoes/octopoes/models/ooi/findings.py [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
     id: str
 
+    description: Optional[str]
+    source: Optional[AnyUrl]
+
+    risk_score: Optional[float]
+    risk_severity: Optional[RiskLevelSeverity]
```
Comment (Donnype) [inline octopoes/octopoes/models/ooi/findings.py:26]: Do these make more sense on a FindingType versus on a Finding? I can imagine two Findings having a different risk_severity depending on which object they point to. (E.g. A `RetireJSFindingType` on your main website has higher risk than on third party websites found by KAT.) In that case risk attributes on the FindingTypes could also be interpreted as defaults.

But then again I have not gone into the risk modelling that much yet so far.

**Review by Donnype — COMMENTED** [review]
Are these

File: octopoes/octopoes/models/ooi/findings.py (hunk 2 above)
Comment (dekkers) [reply octopoes/octopoes/models/ooi/findings.py:26]: Findings like the KAT findings currently have a static score that is used for all the findings that are created and we need to have a place to store those. They are currently stored in the rocky database, but we are moving those to xtdb and thus need the fields also there.

Comment (Donnype) [reply octopoes/octopoes/models/ooi/findings.py:26]: Yes seems legitimate, but I was just questioning the fact that all Findings of the same type have the same risk score and severity as there is no way to differentiate the risk between two findings of the same type now.

Comment (Lisser) [reply octopoes/octopoes/models/ooi/findings.py:26]: @Donnype 

I think your point is valid, a Finding of a certain FindingType might be more or less severe depending on its relation to the Organisation. However, this is a separate use-case and needs more thought on how, when and where to implement this functionality. 

For now, as @dekkers says, this story concerns 'just' moving the FindingType metadata to XTDB/Octopoes to improve performance of several views and the reports.

File: octopoes/octopoes/models/ooi/findings.py (hunk 1 above)
Comment (Lisser) [reply octopoes/octopoes/models/ooi/findings.py]: Fixed

**Review by ammar92 — APPROVED** [review]

**Review by underdarknl — APPROVED** [review]

**Pull request merged**